sycl/include/CL/sycl/detail/cg.hpp

```diff
@@ -36,18 +36,66 @@ namespace sycl {
 
 // Forward declarations
 class queue;
-namespace detail {
-class queue_impl;
-} // namespace detail
 
 namespace detail {
 
+// Used to represent a type of an extended member
+enum class ExtendedMembersType: unsigned int {
+  HANDLER_KERNEL_BUNDLE = 0,
+};
+
+// Holds a pointer to an object of an arbitrary type and an ID value which
+// should be used to understand what type pointer points to.
+// Used as to extend handler class without introducing new class members which
+// would change handler layout.
+struct ExtendedMemberT {
```

Comment:
As I can guess, this is aimed at solving some handler class backward compatibility problem. Is there some more detailed description elsewhere of the problem and solution? If not, it would be highly useful for the review and further maintenance to have it here. I don't fully understand what e.g. `convertToExtendedMembers` does.

Few more questions which would be good to address in comments:
- I assume there is some correspondence between a CG version and the set of extended members
- I assume changing the set of extended members should lead to CG version change

Reply:
Added a comment.